### DISCUSSION TITLE
Add device "Control serial number if exist"

### DISCUSSION THREAD

**Discussion by alex001x** [opendcim/openDCIM#1503] [Ideas]
It would be interesting to have a button or a validation function when adding equipment that checks if the equipment's serial number already exists in the database, in order to avoid duplication. I would like to assist the OpenDCIM team and look forward to hearing from you.
![device function if exist serial nb in database](https://github.com/opendcim/openDCIM/assets/146840804/d5948b2d-a477-4751-ab42-ad3e845b8078)

**Comment by wilpig** [comment]
No.  That format looks like a data entry problem and there would be no way to differentiate it from an actual error and someone would assume this might prevent two devices from getting entered with the same serial number.  There is nothing stating that a serial number can't be "1" from cisco and "1" from hp and both be valid serial numbers.  

If you wanted to build out a report that could show duplicates in the database, that would be acceptable and we'd accept a pull request on that, but this will NOT even be considered the the UI.

**Comment by alex001x** [comment]
Hi Wilpig
Thks, for you idea " creation of a report that can display duplicate equipment name and serial number in the database
You can make a pull request please. Thank you
Alex.
